optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/lookup/ClassAndPlanningIdComparator.java

```diff
@@ -21,18 +21,25 @@
 import java.util.Map;
 
 import org.optaplanner.core.api.domain.lookup.PlanningId;
+import org.optaplanner.core.config.solver.DomainAccessType;
 import org.optaplanner.core.config.util.ConfigUtils;
 import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
 
 public class ClassAndPlanningIdComparator implements Comparator<Object> {
 
     private boolean failFastIfNoPlanningId;
+    private DomainAccessType domainAccessType;
 
     public ClassAndPlanningIdComparator() {
-        this(true);
+        this(DomainAccessType.REFLECTION, true);
```

Comment:
Add `// TODO This will break Quarkus once we don't open up the domain hierarchy for reflection any more`